iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/optional.inl

```diff
@@ -24,11 +24,15 @@ namespace iox
 {
 namespace cxx
 {
+// AXIVION DISABLE STYLE AutosarC++19_03-A12.6.1 : field intentionally not initialized
+// AXIVION DISABLE STYLE AutosarC++19_03-A13.3.1 : that is constrained (via SFINAE) to not match
+// any other overloads
+// AXIVION Next Construct AutosarC++19_03-A0.1.4 : TODO
 // m_data is not initialized since this is a constructor for an optional with no value; an access of the value would
 // lead to the application's termination
 // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init, hicpp-member-init)
 template <typename T>
-inline optional<T>::optional(const nullopt_t& noVal IOX_MAYBE_UNUSED) noexcept
+inline optional<T>::optional(const nullopt_t noVal IOX_MAYBE_UNUSED) noexcept
```

Comment:
When we remove the variable name we can also get rid of `IOX_MAYBE_UNUSED` and the code would become more readable.